components/sup/doc/http_gateway_services_schema.json

```diff
@@ -0,0 +1,521 @@
+      "description": "A Habitat package",
+      "required": [
+        "name",
+        "origin",
+        "release",
+        "version"
+      ],
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "The package name"
+        },
+        "origin": {
+          "type": "string",
+          "description": "The package origin"
+        },
+        "release": {
+          "type": "string",
+          "description": "The package release"
+        },
+        "version": {
+          "type": "string",
+          "description": "The package version"
+        }
+      }
+    }
+  },
+  "items": {
+    "type": "object",
+    "description": "The services that the Habitat supervisor has loaded",
+    "required": [
+      "all_pkg_binds",
+      "binding_mode",
+      "binds",
+      "bldr_url",
+      "cfg",
+      "channel",
+      "config_from",
+      "desired_state",
+      "health_check",
+      "hooks",
+      "initialized",
+      "last_election_status",
+      "manager_fs_cfg",
+      "needs_reconfiguration",
+      "needs_reload",
+      "pkg",
+      "process",
+      "service_group",
+      "smoke_check",
+      "spec_file",
+      "spec_ident",
+      "sys",
+      "topology",
+      "update_strategy",
+      "user_config_updated"
+    ],
+    "properties": {
+      "all_pkg_binds": {
+        "type": "array",
+        "description": "All of the package bind information",
+        "items": {
+          "type": "object",
+          "description": "A package bind",
+          "required": [
+            "exports",
+            "service"
+          ],
+          "properties": {
+            "exports": {
+              "type": "array",
+              "description": "The binds that the service exports",
+              "items": {
+                "type": "string",
+                "description": "The bind value"
+              }
+            },
+            "service": {
+              "type": "string",
+              "description": "The bind name"
+            }
+          }
+        }
+      },
+      "binding_mode": {
+        "type": "string",
+        "description": "What type of bind this is, either strict or relaxed"
+      },
+      "binds": {
+        "type": "array",
+        "description": "Mapping of bind name to service group",
+        "items": {
+          "type": "string",
+          "description": "A colon separated list of two items: the bind name, and service group",
+          "examples": [
+            "router:builder-router.default"
+          ]
+        }
+      },
+      "bldr_url": {
+        "type": "string",
+        "description": "The URL for the Habitat Builder service"
+      },
+      "cfg": {
+        "type": "object",
+        "description": "The current configuration of this service",
+        "additionalProperties": {
+          "type": ["boolean", "string", "integer", "array", "object"]
+        }
+      },
+      "channel": {
+        "type": "string",
```

Comment:
Shouldn't this be nullable? Not every service is configured to update.

Reply:
It's not optional [in the struct](https://github.com/habitat-sh/habitat/blob/master/components/sup/src/manager/service/mod.rs#L98)

Reply:
Whoa, that's interesting 🤔 

I bet we just end up running with a default value if not otherwise set, and then whether that actually does anything is governed by the update strategy.

That would be interesting to clean up at some point (not here, though).